In [1]:
import os
import pandas as pd
import csv

In [2]:
#Id1 == concept_label id2 == parent_label
os.chdir("../wordnet")
tab = pd.read_csv("noun_closure.csv")
tab.head()

FileNotFoundError: [Errno 2] File noun_closure.csv does not exist: 'noun_closure.csv'

In [9]:
#level one
lvl_one = tab[tab["id2"] == "entity.n.01" ]
lvl_one
erg = list(lvl_one.id1)
erg

['weather_bureau.n.01',
 'decalcomania.n.02',
 'bandage.n.01',
 'pennywhistle.n.01',
 'galantine.n.01',
 'passionflower.n.01',
 'sex.n.02',
 'pencil.n.02',
 'donkey.n.01',
 'elasmobranch.n.01',
 'preservation.n.01',
 'cabinet.n.02',
 'azote.n.01',
 'condescension.n.02',
 "mason's_level.n.01",
 'chronic_obstructive_pulmonary_disease.n.01',
 'coaling_station.n.01',
 'new_world_tapir.n.01',
 'initiate.n.02',
 'slender_wild_oat.n.01',
 'engraulidae.n.01',
 'notion.n.02',
 'typewriter_paper.n.01',
 'flying_gecko.n.01',
 'chin_strap.n.01',
 'cocarboxylase.n.01',
 'pirate.n.02',
 'directory.n.02',
 'spear.n.01',
 'peptization.n.01',
 'tetryl.n.01',
 'bongo.n.02',
 'worth.n.01',
 'greek_orthodox_church.n.01',
 'lemmus.n.01',
 'diegueno.n.02',
 'god.n.03',
 "achilles'_heel.n.01",
 'diploid.n.01',
 'theta_rhythm.n.01',
 'fraction.n.02',
 'coir.n.01',
 'unsnarling.n.01',
 'potato_skin.n.01',
 'conservancy.n.01',
 'advantage.n.02',
 'existentialist.n.01',
 "saint_anthony's_fire.n.01",
 'v-j_day.n.

In [19]:
import random 
rand_targets = random.choices(erg, k=10)
print(rand_targets)

['humorist.n.01', 'substance.n.01', 'leeward_tide.n.01', 'effeminacy.n.01', 'employable.n.01', 'spastic_bladder.n.01', 'genus_apteryx.n.01', 'root_rot.n.01', 'gestalt.n.01', 'phlebotomus.n.02']


In [23]:
#get children of random targets 

rand_t_children = []
for n in rand_targets:
    plot_label = tab[tab["id1"]  == n]["id2"]
    rand_t_children.append(plot_label.any()) 
    
    
print(rand_t_children)

['physical_entity.n.01', 'matter.n.03', 'event.n.01', 'trait.n.01', 'object.n.01', 'bladder_disorder.n.01', 'abstraction.n.06', 'abstraction.n.06', 'form.n.03', 'abstraction.n.06']


In [26]:
#level event
lvl_ev = tab[tab["id2"] == "event.n.01" ]
lvl_ev
erg = list(lvl_ev.id1)
erg

['interplay.n.01',
 'bow-wow.n.01',
 'anthropometry.n.01',
 'nonconformism.n.02',
 '9/11.n.01',
 'gravure.n.04',
 'census.n.01',
 'cautery.n.02',
 'major_surgery.n.01',
 'check.n.12',
 'disservice.n.01',
 'curtain_raiser.n.02',
 'delegacy.n.03',
 'tolerance.n.03',
 'lacrosse.n.01',
 'metaurus_river.n.01',
 'lace_making.n.01',
 'declamation.n.02',
 'injustice.n.01',
 'boxing.n.01',
 'electronic_surveillance.n.01',
 'anti-takeover_defense.n.01',
 'accreditation.n.01',
 'indoctrination.n.01',
 'violence.n.01',
 'rollover.n.01',
 'traffic_control.n.01',
 'prognosis.n.02',
 'infantilism.n.02',
 'power_walking.n.01',
 'graduation.n.01',
 'concentrated_fire.n.01',
 'failure.n.02',
 'stopping.n.02',
 'prayer.n.01',
 'cotillion.n.02',
 'jumpstart.n.01',
 'kowtow.n.01',
 'spare.n.03',
 'hiring_freeze.n.01',
 'republication.n.02',
 'barnburner.n.02',
 'lenience.n.03',
 'drive.n.06',
 'wrongful_death.n.01',
 'sighting.n.01',
 'match_play.n.01',
 'challenge.n.03',
 'confectionery.n.03',
 'game_of_c

In [27]:
len(erg)

8101

In [5]:


#level one
lvl_one = tab[tab["id2"] == "entity.n.01" ]
lvl_one
erg = list(lvl_one.id1)
erg
len(erg)

82096

In [3]:
#find multi parent rels
del_select = tab
#level one
multi_par = del_select[del_select['id1'].duplicated() ]
multi_par
#erg = list(map(int,lvl_one.concept_id))
#erg

#lvl_one.to_csv('all_multi_parent_children.csv',header=True,sep=',',index = False)


id1                   id2  weight
478             lesser_celandine.n.01    spermatophyte.n.01       1
521                          bug.n.02            state.n.02       1
680                       kansas.n.01           region.n.03       1
801               rosebud_orchid.n.01           flower.n.01       1
885              winter_flounder.n.02     teleost_fish.n.01       1
...                               ...                   ...     ...
743081                     sloth.n.01           entity.n.01       1
743082                  agrostis.n.01  taxonomic_group.n.01       1
743083         corporate_finance.n.01              act.n.02       1
743084  revolving_charge_account.n.01           assets.n.01       1
743085                      fica.n.01      liabilities.n.01       1

[660972 rows x 3 columns]

In [4]:
#look which multi-parent kids have kinds on their own

for index, row in multi_par.iterrows():
    target = row['id1']
    kids = tab[tab['id2'] == target]
    if len(kids) >= 3:
        print('{} at index {} has {} children'.format(target,index,len(kids)))

preservation.n.01 at index 1066 has 9 children
vole.n.01 at index 1668 has 8 children
philanthropist.n.01 at index 2116 has 13 children
neurotransmitter.n.01 at index 2209 has 3 children
milkweed.n.01 at index 2256 has 10 children
scarf.n.01 at index 2328 has 16 children
curtain.n.01 at index 2870 has 7 children
transmission.n.01 at index 2967 has 6 children
oligosaccharide.n.01 at index 3175 has 9 children


KeyboardInterrupt: 

#multi-parent children
preservation.n.01 at index 1066 has 9 children

philanthropist.n.01 at index 2116 has 13 children

compiler.n.02 at index 4516 has 4 child

old_english.n.01 at index 5805 has 3 children

fatty_acid.n.01 at index 7107 has 27 children

path.n.04 at index 9734 has 8 children

In [13]:
#look which multi-parent kids have kinds on their own

for index, row in multi_par.iterrows():
    target = row['id2']
    kids = tab[tab['id1'] == target]
    if len(kids) >= 3:
        print('{} at index {} has {} parents'.format(target,index,len(kids)))

spermatophyte.n.01 at index 478 has 8 parents
state.n.02 at index 521 has 3 parents
region.n.03 at index 680 has 4 parents
flower.n.01 at index 801 has 10 parents
teleost_fish.n.01 at index 885 has 12 parents
chordate.n.01 at index 1009 has 7 parents
part.n.01 at index 1030 has 3 parents
act.n.02 at index 1066 has 4 parents
movie.n.01 at index 1106 has 12 parents
organism.n.01 at index 1381 has 5 parents
reproductive_structure.n.01 at index 1384 has 7 parents
living_thing.n.01 at index 1516 has 4 parents
lawman.n.01 at index 1603 has 10 parents
poet.n.01 at index 1615 has 10 parents
placental.n.01 at index 1668 has 10 parents
living_thing.n.01 at index 1688 has 4 parents
pakistani_monetary_unit.n.01 at index 1689 has 6 parents
plant.n.02 at index 1813 has 6 parents
creator.n.02 at index 1832 has 8 parents
yuman.n.01 at index 1908 has 7 parents
whole.n.02 at index 2116 has 3 parents
compound.n.02 at index 2209 has 9 parents
living_thing.n.01 at index 2247 has 4 parents
plant.n.02 at ind

KeyboardInterrupt: 


living_thing.n.01 at index 1688 has 4 parents

plant.n.02 at index 2256 has 6 parents

state.n.02 at index 521 has 3 parents

placental.n.01 at index 1668 has 10 parents

carnivore.n.01 at index 3540 has 11 parents

part.n.01 at index 1030 has 3 parents

act.n.02 at index 1066 has 4 parents

In [5]:
living_thing_parents = tab[tab['id1'] == "living_thing.n.01"]

plant_parents = tab[tab['id1'] == "plant.n.02"]

state_parents = tab[tab['id1'] == "state.n.02"]

placental_parents = tab[tab["id1"] == "placental.n.01"]

#print(living_thing_parents,"\n\n", plant_parents, "\n\n", state_parents, "\n\n", placental_parents )

print(list(living_thing_parents.id2),"\n\n", list(plant_parents.id2), "\n\n", list(state_parents.id2), "\n\n", list(placental_parents.id2))

['entity.n.01', 'physical_entity.n.01', 'object.n.01', 'whole.n.02'] 

 ['object.n.01', 'whole.n.02', 'entity.n.01', 'organism.n.01', 'physical_entity.n.01', 'living_thing.n.01'] 

 ['attribute.n.02', 'abstraction.n.06', 'entity.n.01'] 

 ['mammal.n.01', 'object.n.01', 'living_thing.n.01', 'chordate.n.01', 'entity.n.01', 'organism.n.01', 'vertebrate.n.01', 'whole.n.02', 'physical_entity.n.01', 'animal.n.01']


In [31]:
#children of those four multi-Parent targets

living_thing_kids = tab[tab['id2'] == "living_thing.n.01"].head()

plant_kids = tab[tab['id2'] == "plant.n.02"].head()

state_kids = tab[tab['id2'] == "state.n.02"].head()

placental_kids = tab[tab["id2"] == "placental.n.01"].head()

#print(living_thing_kids,"\n\n", plant_kids, "\n\n", state_kids, "\n\n", placental_kids )

print(list(living_thing_kids.id1),"\n\n", list(plant_kids.id1), "\n\n", list(state_kids.id1), "\n\n", list(placental_kids.id1))

['leafy_spurge.n.01', 'reed_meadow_grass.n.01', 'strymon_melinus.n.01', 'rosewood.n.02', 'misogynist.n.01'] 

 ["florist's_gloxinia.n.01", 'small_cane.n.01', 'wild_pansy.n.01', 'ditch_reed.n.01', 'kohleria.n.01'] 

 ['anaplasia.n.01', 'distress.n.02', 'bug.n.02', 'repute.n.01', 'varus.n.01'] 

 ['farm_horse.n.01', 'mediterranean_water_shrew.n.01', 'eastern_chimpanzee.n.01', 'vole.n.01', 'false_saber-toothed_tiger.n.01']


In [10]:
#get parents of leafy spurge and misogynist

print(tab[tab['id1'] == "rosewood.n.02"].id2)
print("\n\n")
print(tab[tab['id1'] == "misogynist.n.01"].id2)

164          living_thing.n.01
6873             organism.n.01
212112               tree.n.01
316344             object.n.01
432232    physical_entity.n.01
589780             entity.n.01
644550        woody_plant.n.01
667152              whole.n.02
693350     vascular_plant.n.01
708045              plant.n.02
Name: id2, dtype: object



223            living_thing.n.01
42329          causal_agent.n.01
137730    unpleasant_person.n.01
199398               person.n.01
233886          misanthrope.n.01
357821                whole.n.02
380133             organism.n.01
399499     unwelcome_person.n.01
574773               entity.n.01
625095      physical_entity.n.01
681544               object.n.01
709371               grouch.n.01
Name: id2, dtype: object


In [ ]:
#Target list for plotting

['entity.n.01', 'physical_entity.n.01', 'object.n.01', 'whole.n.02','object.n.01', 'whole.n.02', 'entity.n.01', 'organism.n.01', 'physical_entity.n.01', 'living_thing.n.01','attribute.n.02', 'abstraction.n.06', 'entity.n.01','mammal.n.01', 'object.n.01', 'living_thing.n.01', 'chordate.n.01', 'entity.n.01', 'organism.n.01', 'vertebrate.n.01', 'whole.n.02', 'physical_entity.n.01', 'animal.n.01']


['leafy_spurge.n.01', 'reed_meadow_grass.n.01', 'strymon_melinus.n.01', 'rosewood.n.02', 'misogynist.n.01',

 "florist's_gloxinia.n.01", 'small_cane.n.01', 'wild_pansy.n.01', 'ditch_reed.n.01', 'kohleria.n.01' ,'anaplasia.n.01', 'distress.n.02', 'bug.n.02', 'repute.n.01', 'varus.n.01','farm_horse.n.01', 'mediterranean_water_shrew.n.01', 'eastern_chimpanzee.n.01', 'vole.n.01', 'false_saber-toothed_tiger.n.01']